libyul/backends/evm/EVMMetrics.cpp

```diff
@@ -120,5 +122,7 @@ void GasMeterVisitor::instructionCostsInternal(evmasm::Instruction _instruction)
 		m_runGas += evmasm::GasCosts::keccak256Gas + evmasm::GasCosts::keccak256WordGas;
 	else
 		m_runGas += evmasm::GasMeter::runGas(_instruction, m_dialect.evmVersion());
-	m_dataGas += singleByteDataGas();
+
+	if (_instruction != evmasm::Instruction::PUSH0)
+		m_dataGas += singleByteDataGas();
```

Comment:
Also here: there is still a single byte of data for the ``PUSH0`` opcode itself, isn't there? In that regard it's not different from any other instruction.

Reply:
Fixed.